src/controllers/bundler.controller.ts

```diff
@@ -0,0 +1,40 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import { ProblemException } from '../exceptions/problem.exception';
+import { Controller } from '../interfaces';
+import bundler from '@asyncapi/bundler';
+import Document from '@asyncapi/bundler/lib/document.js';
+import { documentValidationMiddleware } from '../middlewares/document-validation.middleware';
+import path from 'path';
+
+export class BundlerController implements Controller {
+  public basepath = '/bundle';
+  private async bundle(req: Request, res: Response, next: NextFunction) {
+    const asyncapis: Array<string> = req.body.asyncapis || [req.body.asyncapi];
+    if (!asyncapis || !asyncapis.length) {
+      throw new ProblemException({
+        type: 'https://api.asyncapi.com/problem/asyncapi-missing',
+        title: 'Asyncapi missing',
+        detail: 'No Asyncapi provided',
+        status: 400,
+      });
+    }
+    try {
+      const document = await bundler.bundle(asyncapis, { base: path.dirname(asyncapis[0]) });
+      const doc = new Document(document).json();
```

Comment:
`bundle` function returns Document instance -> https://github.com/asyncapi/bundler/blob/master/lib/index.js#L45

You cannot use `path` and filesystem because it's a server, `asyncapis` field contains AsyncAPI documents, not paths for them, in this same way, `base` field is AsyncAPI document sent as JSON/YAML, not as path.

```suggestion
      const document = await bundler.bundle(this.body.asyncapis, { base: req.body.base });
      const doc = document.json();
```